File: 0070. Climbing Stairs/Solution0070.js
/**
 * @param {number} n
 * @return {number}
 */
 var climbStairs = function(n) {
    if (n < 3) return n;
    let pp = 1;
    let p = 1;
    let tmp = 2;
    for (let i = 2; i <= n; i++) {
        tmp = pp + p;
        pp = p;
        p = tmp;
    }
    return tmp;
};

File: 0070. Climbing Stairs/Solution0070.cpp
class Solution0070 {
public:
    int climbStairs(int n) {
        if (n < 2) return n;
        int pp = 1, p = 1, cur = 2;
        for (int i = 2; i <= n; i++) {
            cur = pp + p;
            pp = p; 
            p = cur;
        }
        return cur;
    }
};